images/base/Dockerfile

```diff
@@ -86,10 +86,12 @@ RUN echo "Installing Packages ..." \
     && echo "ReadKMsg=no" >> /etc/systemd/journald.conf \
     && ln -s "$(which systemd)" /sbin/init
 
-RUN echo "Enabling services ... " \
+# NOTE: systemd-binfmt.service will register things into binfmt_misc which is kernel-global
+RUN echo "Enabling / Disabling services ... " \
     && systemctl enable kubelet.service \
     && systemctl enable containerd.service \
-    && systemctl enable undo-mount-hacks.service
+    && systemctl enable undo-mount-hacks.service \
+    && systemctl mask systemd-binfmt.service
```

Comment:
it seems something happened with this in bookworm, a quick search on google show several hits https://github.com/microsoft/WSL/issues/8843 

AFAIK this reads from a folder, can we have any side effect of doing this hammer approach or should we try to be more precise and understand what is exactly breaking this?

Reply:
oh, I see, it screws the list entirely https://github.com/kubernetes-sigs/kind/issues/3510

Reply:
I imagine then that the best way is to say the docker container must use the one existing in the host

/lgtm

Reply:
so often binfmt_misc is intentionally set by containers (see e.g. our multi-arch build setup), but kind nodes shouldn't be doing it, and python scripts should keep using shebangs and not depend on this anyhow ...

Reply:
Per https://systemd.io/CONTAINER_INTERFACE/ (execution environment number 2) `/proc/sys` being a read-only mount is the recommended way to tell systemd not to change things. However as that mentions also making `/proc/sys/net` writable may be desirable (and likely would be needed for many k8s workloads).

It likely needs more thought than will fit in a review comment, so this targeted workaround makes sense for now.

Reply:
Yeah, the base image comments link to this page -- We have other similar workarounds for similar reasons, we're really stretching "systemd in a container" when we go `--privileged` but without it we can't get the "kubernetes 'in' a container" part. Previously in practice the key part was `/sys` mounted ro (mentioned further down as a trigger for udev) which we do.